### PR TITLE
whisperkit-cli 0.4.0 (new formula)

### PR DESCRIPTION
**Pull request by jkrukowski**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/w/whisperkit-cli.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  homepage "https://github.com/argmaxinc/WhisperKit"
+  url "https://github.com/argmaxinc/WhisperKit.git",
+      tag:      "v0.4.0",
+      revision: "59cb8516c708e3e2f18198002600026b5a1135ca"
+  license "MIT"
+
+  depends_on xcode: ["15.0", :build]
+  depends_on arch: :arm64
+  depends_on :macos
+  depends_on macos: :ventura
```
[the remark this answers is not in the document]
Comment (jkrukowski) [reply Formula/w/whisperkit-cli.rb:12]: we use Apple's Neural Engine to run the inference when using `whisperkit-cli`, we have an option to run on cpu as well, let me try it

[the remark this answers is not in the document]
Comment (jkrukowski) [reply Formula/w/whisperkit-cli.rb:12]: we use Apple's Neural Engine to run the inference when using `whisperkit-cli`, we have an option to run on cpu as well, let me try it

[the remark this answers is not in the document]
Comment (jkrukowski) [reply Formula/w/whisperkit-cli.rb:12]: it throws an error on x86 so we have to restrict it to arm64 only

[the remark this answers is not in the document]
Comment (p-linnane) [reply Formula/w/whisperkit-cli.rb:12]: I think it's fine to accept this. We're 6 months out from macOS 15 anyway, and then this should support all arm64 versions.

File: Formula/w/whisperkit-cli.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      "tiny",
+      "--download-model-path",
+      "#{testpath}/model",
+      "--download-tokenizer-path",
+      "#{testpath}/tokenizer",
+      "--audio-path",
+      test_fixtures("test.mp3"),
+      "--verbose",
+    ].join(" ")
+    assert_includes shell_output(whisperkit_command), "Transcription:"
```
Comment (MikeMcQuaid) [inline Formula/w/whisperkit-cli.rb:36]:
```suggestion
    ]
    assert_includes shell_output(*whisperkit_command), "Transcription:"
```

I think this should work and it's preferable to pass these things as separate arguments when possible.

**Review by MikeMcQuaid — COMMENTED** [review]
One nit (which may not even be correct): then 👍🏻  from me.

File: Formula/w/whisperkit-cli.rb (hunk 2 above)
Comment (jkrukowski) [reply Formula/w/whisperkit-cli.rb:36]: looks like this is throwing an error
```
  An exception occurred within a child process:
    ArgumentError: wrong number of arguments (given 11, expected 1..2)
```

Comment (jkrukowski) [reply Formula/w/whisperkit-cli.rb:36]: without the `*` it's throwing 

```
 sh: [/opt/homebrew/Cellar/whisperkit-cli/0.4.0/bin/whisperkit-cli,: No such file or directory
```

Comment (jkrukowski) [reply Formula/w/whisperkit-cli.rb:36]: so I reverted to the original one @MikeMcQuaid

**Review by MikeMcQuaid — APPROVED** [review]
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @jkrukowski!

**Review by p-linnane — APPROVED** [review]

**Pull request merged**